genie-web/src/main/java/com/netflix/genie/web/rpc/grpc/services/impl/v4/GRpcJobFileServiceImpl.java

```diff
@@ -0,0 +1,563 @@
+/*
+ *
+ *  Copyright 2019 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.web.rpc.grpc.services.impl.v4;
+
+import com.google.common.collect.Sets;
+import com.google.protobuf.Empty;
+import com.netflix.genie.common.internal.dto.JobDirectoryManifest;
+import com.netflix.genie.common.internal.dto.v4.converters.JobDirectoryManifestProtoConverter;
+import com.netflix.genie.common.internal.exceptions.GenieConversionException;
+import com.netflix.genie.common.internal.exceptions.StreamUnavailableException;
+import com.netflix.genie.common.internal.util.FileBuffer;
+import com.netflix.genie.common.internal.util.FileBufferFactory;
+import com.netflix.genie.proto.AgentJobFileMessage;
+import com.netflix.genie.proto.FileStreamAck;
+import com.netflix.genie.proto.FileStreamChunk;
+import com.netflix.genie.proto.FileStreamRegistration;
+import com.netflix.genie.proto.FileStreamRequest;
+import com.netflix.genie.proto.JobFileManifestMessage;
+import com.netflix.genie.proto.JobFileServiceGrpc;
+import com.netflix.genie.proto.ServerJobFileMessage;
+import com.netflix.genie.web.properties.GRpcServerProperties;
+import com.netflix.genie.web.rpc.grpc.interceptors.SimpleLoggingInterceptor;
+import com.netflix.genie.web.services.AgentFileManifestService;
+import com.netflix.genie.web.services.AgentFileStreamService;
+import io.grpc.stub.StreamObserver;
+import lombok.extern.slf4j.Slf4j;
+import net.devh.springboot.autoconfigure.grpc.server.GrpcService;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+
+import javax.annotation.Nullable;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Path;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Implementation of {@link JobFileServiceGrpc.JobFileServiceImplBase} to serve files via API for jobs being executed
+ * by a remote agent connected through gRPC.
+ *
+ * @author mprimi
+ * @since 4.0.0
+ */
+@ConditionalOnProperty(value = GRpcServerProperties.ENABLED_PROPERTY, havingValue = "true")
+@GrpcService(
+    value = JobFileServiceGrpc.class,
+    interceptors = {
+        SimpleLoggingInterceptor.class,
```

Comment:
Well... depends on what you mean by use.
Having a log of all gRPC calls could be pretty useful (i.e. in case something is hammering the servers) at least until we have metrics in place.